### ISSUE TITLE
Create a websocket server enabled Botkit connector which can manage multiple remote clients

### ISSUE THREAD

**Issue by benbrown** [howdyai/botkit#871]

**Comment by benbrown** [comment]
The code for this is available in this pull request: 
https://github.com/howdyai/botkit/pull/872

I've also created a simple web client starter kit:
https://github.com/howdyai/botkit-starter-web

**Comment by munawwarhussain** [comment]
can we use this library https://github.com/primus/primus to that will let the use se any socket library.

**Comment by benbrown** [comment]
@munawwarhussain That is very interesting! I'll have to take a look.  What are the alternative socket libraries you are interested in?

**Comment by maxmeinberg** [comment]
@benbrown we are waiting for something like this!!!

Is it possible to use the Web Bot without a Botkit Studio token, running in my own server?

**Comment by peterswimm** [comment]
@maxmeinberg Yes! You can run it standalone like any other integration, but studio will greatly extend it's capabilities.

**Comment by munawwarhussain** [comment]
@benbrown problem with socket.io is that it has problem scaling and good to have options open.

**Comment by maxmeinberg** [comment]
@benbrown I'm trying to use this Web Bot with my existing Facebook bot and got this error message 
"Cannot read property 'middleware' of undefined" when call Web.controller.middleware.receive.use(middleware.receive);

Is it possible to use the Web Bot in same server with Facebook Bot and Watson?

This is part of my code:
module.exports = function(app) {
  if (process.env.USE_WEB) {
    var Web = require('./bot-web');
    Web.controller.middleware.receive.use(middleware.receive);
    console.log('Web bot is live');
  }
  if (process.env.USE_FACEBOOK) {
    var Facebook = require('./bot-facebook');
    Facebook.controller.middleware.receive.use(middleware.receive);
    Facebook.controller.createWebhookEndpoints(app, Facebook.bot);
    console.log('Facebook bot is live');
  }
  
  // Customize your Watson Middleware object's before and after callbacks.
  middleware.before = function(message, conversationPayload, callback) {
    if (message.user == process.env.FACE_PAGE_ID)
      callback(null, conversationPayload);
    else
      callback(true);
  }

  middleware.after = function(message, conversationResponse, callback) {
    console.log('After message', message);
    callback(null, conversationResponse);
  }
};

Thank you very much!!

**Comment by tmundt** [comment]
@maxmeinberg That should work, as you pass in middleware depending on Node's ENV variable setting! Would be interesting how you handle of different tokens depending on the Bot's realm (that is Facebook, Slack and the like).

**Comment by gagan-bansal** [comment]
+1
